File: solutions/cpp/hamming/hamming.cpp
#include "hamming.h"

namespace hamming
{
    using namespace std;

    int compute(string const &a, string const &b)
    {
        if (a.size() != b.size())
        {
            throw domain_error("different lengths");
        }

        // Calculate the Hamming distance between two DNA strands.
        int difference_count{0};
        for (size_t i = 0; i < a.size(); i++)
        {
            if (a[i] != b[i])
            {
                difference_count += 1;
            }
        }
        return difference_count;
    }

} // namespace hamming

File: solutions/lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
    if #a ~= #b then
        return -1 -- different lengths
    end

    -- Calculate the Hamming Distance between two DNA strands.
    local distance = 0
    for i = 0, #a do
        if a:sub(i, i) ~= b:sub(i, i) then
            distance = distance + 1
        end
    end
    return distance
end

return Hamming
